commit c199fff7402f6872bf2f34c7a5648bd6e0e26d0f
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Mon Feb 6 23:09:48 2017 +0800

    Fix focus problem in Arch
    
    According to Qt docs [1], Qt::X11BypassWindowManagerHint leads to "no
    keyboard input unless you call QWidget::activateWindow() manually".
    
    Calling activateWindow() here fixes the focus problem [2] in Arch.
    
    [1] http://doc.qt.io/qt-5/qt.html
    [2] https://github.com/linuxdeepin/developer-center/issues/156
    
    Change-Id: Ifc80b50d008a025ef68f760cfa2741f37a230989

diff --git a/frame/frame.cpp b/frame/frame.cpp
index acb2c351..ceef7fe6 100644
--- a/frame/frame.cpp
+++ b/frame/frame.cpp
@@ -220,6 +220,7 @@ void Frame::show()
 
     // show frame
     QFrame::show();
+    QFrame::activateWindow();
 
     // notify top widget appear
     if (m_frameWidgetStack.last() && m_frameWidgetStack.last()->content())
